src/main/java/gift/controller/KakaoController.java

```diff
@@ -0,0 +1,44 @@
+package gift.controller;
+
+import gift.service.KakaoService;
+import java.net.URI;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class KakaoController {
+
+    private final KakaoService service;
+
+    @Value("${kakao.api.key}")
+    private String apiKey;
+
+    @Value("${redirect.url}")
+    private String redirectUrl;
+
+    public KakaoController(KakaoService service) {
+        this.service = service;
+    }
+
+    @GetMapping("/oauth/login")
+    public ResponseEntity<Void> redirectToKakao() {
+        URI uri = URI.create("https://kauth.kakao.com/oauth/authorize" +
+                "?response_type=code" +
+                "&client_id=" + apiKey +
+                "&redirect_uri="+ redirectUrl);
```

Comment:
```suggestion
URI uri = UriComponentsBuilder.fromUriString("https://kauth.kakao.com")
                .path("/oauth/authorize")
                .queryParam("response_type", "code")
                .queryParam("client_id", apiKey)
                .queryParam("redirect_uri", redirectUrl)
                .build().toUri();
```
URI를 만들 때는 builder를 활용하면 구조적으로 만들 수 있어 더 명확하고 안전하게 생성이 가능합니다!

Reply:
그리고 URI 생성은 KakaoService 에 위임하는 것이 어떨까요? 
카카오 로그인에 대한 로직 관리는 서비스가 적절해 보이고, apiKey, redirectUrl 관리도 한 곳에서 할 수 있습니다!
+ 추가로 이 uri 는 앱이 실행된 후에는 변하지 않을 것 같은데 매번 요청 때마다 생성하지 않아도 될 것 같아요!